stan/math/prim/mat/functor/map_rect_concurrent.hpp

```diff
@@ -5,6 +5,8 @@
 
 #include <stan/math/prim/mat/functor/map_rect_reduce.hpp>
 #include <stan/math/prim/mat/functor/map_rect_combine.hpp>
+#include <boost/lexical_cast.hpp>
+#include <boost/throw_exception.hpp>
```

Comment:
I am not sure why we would need boost's throw... is there any advantage? In any case, I would like to switch to the stan-math functions for throwing standardized error messages, see below.